lib/octokit/client/hooks.rb

```diff
@@ -1,296 +1,159 @@
-      # @option options [Array<String>] :add_events Determines a list of events
-      #   to be added to the list of events that the Hook triggers for.
-      # @option options [Array<String>] :remove_events Determines a list of events
-      #   to be removed from the list of events that the Hook triggers for.
-      # @option options [Boolean] :active Determines whether the hook is
-      #   actually triggered on pushes.
-      # @return [Sawyer::Resource] Hook info for the updated hook
-      # @see https://api.github.com/hooks
-      # @see https://github.com/github/github-services
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @param hook_id [Integer] The ID of the hook
+      # @param config [Object] Key/value pairs to provide settings for this webhook. [These are defined below](https://developer.github.com/v3/repos/hooks/#create-hook-config-params).
+      # @param options [Array] :events Determines what [events](https://developer.github.com/v3/activity/events/types/) the hook is triggered for. This replaces the entire array of events.
+      # @param options [Array] :add_events Determines a list of events to be added to the list of events that the Hook triggers for.
+      # @param options [Array] :remove_events Determines a list of events to be removed from the list of events that the Hook triggers for.
+      # @param options [Boolean] :active Determines if notifications are sent when the webhook is triggered. Set to `true` to send notifications.
+      # @return <Sawyer::Resource> 
       # @see https://developer.github.com/v3/repos/hooks/#edit-a-hook
-      # @example
-      #   @client.edit_hook(
-      #     'octokit/octokit.rb',
-      #     100000,
-      #     'web',
-      #     {
-      #       :url => 'http://something.com/webhook',
-      #       :content_type => 'json'
-      #     },
-      #     {
-      #       :add_events => ['status'],
-      #       :remove_events => ['pull_request'],
-      #       :active => true
-      #     }
-      #   )
-      def edit_hook(repo, id, name, config, options = {})
-        options = {:name => name, :config => config}.merge(options)
-        patch "#{Repository.path repo}/hooks/#{id}", options
+      def edit_hook(repo, hook_id, config = {}, options = {})
+        options[:config] = config
+        raise Octokit::MissingKey.new unless config.key? :url
+        patch "#{Repository.path repo}/hooks/#{hook_id}", options
       end
 
-      # Delete hook
-      #
-      # Requires authenticated client.
+      # Delete a hook
       #
-      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
-      # @param id [Integer] Id of the hook to remove.
-      # @return [Boolean] True if hook removed, false otherwise.
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @param hook_id [Integer] The ID of the hook
+      # @return <Boolean> True on success, false otherwise
```

Comment:
Looks like this should be 

```suggestion
      # @return [Boolean] True on success, false otherwise
```